### ISSUE TITLE
Provide a means of changing client username

### ISSUE THREAD

**Issue by Surgbc** [mochi-co/mqtt#106]
I have come across a situation where it is necessary to change the username used to authenticate the client. I am implementing a system where a user logs in using either (1) username and password (2) webtoken obtained from authentication done by the client itself. Since the `username` user and the `webtoken` user are one and the same, I require to change the client username to userId which is the same regardless of how the client authenticates.

I have so far managed to do it this way:
1. Changing the auth interface
`Authenticate(user, password []byte) (interface{}, error)`
 
2. Obtaining new username and assigning to client during authentication

```cl.Identify(lid, pk, ac) // Set client identity values from the connection packet.

	// if !ac.Authenticate(pk.Username, pk.Password) {
	// 	if err := s.ackConnection(cl, packets.CodeConnectBadAuthValues, false); err != nil {
	// 		return s.onError(cl.Info(), fmt.Errorf("invalid connection send ack: %w", err))
	// 	}
	// 	return s.onError(cl.Info(), ErrConnectionFailed)
	// }
	username, err := ac.Authenticate(pk.Username, pk.Password)
	if err != nil {
		if err := s.ackConnection(cl, packets.CodeConnectBadAuthValues, false); err != nil {
			return s.onError(cl.Info(), fmt.Errorf("invalid connection send ack: %w", err))
		}
		return s.onError(cl.Info(), ErrConnectionFailed)
	}
	if username != nil {
		pk.Username = []byte(username.(string))
		cl.Identify(lid, pk, ac)
	}
```

Is this a feature you could consider?

**Comment by mochi-co** [comment]
Hi @Surgbc - this feature will be available in v2.0.0. V2 is nearly ready (just working out some of the finer details of mqtt v5) 👍🏻

**Comment by mochi-co** [comment]
This issue has been resolved in v2.0.0 through the implementation of universal hooks. You should be able to set the client.ID value in OnConnect or OnConnectAuthenticate as client is a pointer. Let me know if it doesn't behave as expected.

**Comment by Surgbc** [comment]
Great!
